audit/src/main/java/gov/cms/ab2d/audit/cleanup/FileDeletionServiceImpl.java

```diff
@@ -60,6 +62,14 @@ public void deleteFiles() {
         log.info("File deletion service kicked off");
         validateEfsMount();
 
+        File[] files = new File(efsMount).listFiles();
+
+        if (files == null || files.length == 0) {
+            return;
+        }
+
+        List<String> jobIds = Stream.of(files).map(f -> f.getName()).collect(Collectors.toList());
+
         try (Stream<Path> filePaths = Files.walk(Paths.get(efsMount), FileVisitOption.FOLLOW_LINKS)) {
```

Comment:
This stream is still going to cause problems because of concurrent access.

It looks like filePaths.forEach is stepping into directories it needs to ignore. If a job is in progress or submitted you run the risk of these concurrency exceptions no matter what you do.